runtime/commonMain/src/kotlinx/serialization/protobuf/ProtoBuf.kt

```diff
@@ -15,12 +15,96 @@ import kotlinx.serialization.protobuf.ProtoBuf.Varint.decodeSignedVarintLong
 import kotlinx.serialization.protobuf.ProtoBuf.Varint.decodeVarint
 import kotlinx.serialization.protobuf.ProtoBuf.Varint.encodeVarint
 
-class ProtoBuf(context: SerialModule = EmptyModule) : AbstractSerialFormat(context), BinaryFormat {
+/**
+ * The main entry point to work with ProtoBuf serialization.
+ * It is typically used by constructing an application-specific instance, with configured ProtoBuf-specific behaviour
+ * ([encodeDefaults] constructor parameter) and, if necessary, registered
+ * custom serializers (in [SerialModule] provided by [context] constructor parameter).
+ *
+ * ## Usage Example
+ * Given a ProtoBuf definition with one required field, one optional field and one optional field with a custom default
+ * value:
+ * ```
+ * message MyMessage {
+ *     required int32 first = 1;
+ *     optional int32 second = 2;
+ *     optional int32 third = 3 [default = 42];
+ * }
+ * ```
+ *
+ * The corresponding [Serializable] class should match the ProtoBuf definition and should use the same default values:
+ * ```
+ * @Serializable
+ * data class MyMessage(val first: Int, val second: Int = 0, val third: Int = 42)
+ *
+ * // Serialize to ProtoBuf hex string
+ * val encoded = ProtoBuf.dumps(MyMessage(15)) // "080f1000182a"
```

Comment:
Please don't use `@ImplicitReflectionSerializer` methods in the examples, i.e. use `ProtoBuf.dumps(MyMessage.serializer(), MyMessage(15))` / `ProtoBuf.load(MyMessage.serializer(), encoded)` etc